### PR TITLE
[FIX] purchase: set the unit price to zero if requested in the purchase order

### PR DESCRIPTION
**Pull request by nim-odoo**
If a purchase order is created with free goods (e.g. 5 units offered if 100 are purchased),
the price_unit of stock.move and the cost of stock.quant were set to the product cost price
instead of zero. This would lead to inconsistencies in the Current Inventory Valuation.

opw: 630593

### CODE REVIEW

File: addons/purchase/stock.py [hunk 1]
```diff
@@ -31,6 +31,13 @@ class stock_move(osv.osv):
             readonly=True),
     }
 
+    def get_price_unit(self, cr, uid, move, context=None):
+        """ Returns the unit price to store on the quant """
+        if move.purchase_line_id:
+            return move.price_unit
```
Comment (jco-odoo) [inline addons/purchase/stock.py]: @nim-odoo Is not it more logical to use move.price_unit all the time then in the base method?  (I don't know why we kept having twice the same logic in fact)

File: addons/purchase/stock.py [hunk 2]
```diff
@@ -117,6 +124,9 @@ def attribute_price(self, cr, uid, move, context=None):
         """
             Attribute price to move, important in inter-company moves or receipts with only one partner
         """
+        if move.purchase_line_id:
```
Comment (jco-odoo) [inline addons/purchase/stock.py]: Add comment why.

Comment (nim-odoo) [inline addons/purchase/stock.py]: Done

**Comment by jco-odoo** [comment]
merged with commit 34c891a

**Pull request closed without merging**